### PR TITLE
feat: triage policy, coverage ratchet, CLI smoke tests, organizer refactor (#85–#88)

### LINKED ISSUE #85: [FEATURE] Add one-command integration coverage ratchet workflow
**Problem**
The per-module integration coverage ratchet is useful, but the local workflow still feels manual. That makes it easy to fix coverage and forget to refresh the baseline or verify the exact gate shape before merging.

**Goal**
Provide a single repo-sanctioned local command for measuring integration coverage, checking the module floor gate, and optionally ratcheting the baseline upward.

**In scope**
- Review the current flow documented in `scripts/coverage/README.md`
- Add a simple entrypoint for the common local coverage-floor workflow
- Ensure the entrypoint supports both check mode and baseline-update mode
- Document usage and expected outputs

**Out of scope**
- Changing floor policy or tolerance semantics in `scripts/check_module_coverage_floor.py`
- Raising baselines without matching test improvements
- Modifying the GitHub Actions workflow trigger model

**MECE plan**
1. Inventory the current manual steps across `.claude/scripts/measure-integration-coverage.sh`, `scripts/check_module_coverage_floor.py`, and `scripts/coverage/README.md`.
2. Decide on one blessed interface, such as a wrapper script or make-style command.
3. Implement check mode, dry-run ratchet mode, and write mode with clear stdout guidance.
4. Update docs so contributors can reproduce the CI gate locally without guesswork.
5. Verify the new command against a known module-floor scenario.

**Definition of done**
- One documented command covers the common local integration-floor workflow
- Contributors can run check, dry-run, and update flows without stitching commands together manually
- The tool behavior matches CI semantics closely enough to prevent avoidable merge surprises

**Why this issue is separate**
This issue is local developer ergonomics for the existing coverage ratchet. It does not change PR workflow triggers or add new product tests.

### LINKED ISSUE #86: [FEATURE] Refactor organizer core to remove the last complexity exemption
**Problem**
`src/file_organizer/core/organizer.py` is the only remaining Ruff complexity exemption called out in `pyproject.toml`, with `organize` still carrying the `C901` ratchet exception.

**Goal**
Refactor the organizer flow to remove the remaining complexity outlier without changing user-visible behavior.

**In scope**
- Audit `src/file_organizer/core/organizer.py` for the highest-complexity branches
- Extract cohesive helper methods or collaborator objects where that reduces branching pressure
- Preserve current behavior, logging, and error handling
- Add or adjust tests to pin the refactor safely

**Out of scope**
- Broad redesign of the whole pipeline architecture
- CLI UX changes unrelated to maintainability
- CI workflow changes or coverage-baseline tooling

**MECE plan**
1. Map the major decision branches inside `organize` and identify natural seams.
2. Split orchestration, validation, and side-effect-heavy steps into smaller testable units.
3. Remove the Ruff `C901` exception only when the extracted design is stable and readable.
4. Back the refactor with targeted tests around the riskiest branches.
5. Re-run lint, type-check, and the most relevant workflow tests to ensure no behavioral drift.

**Definition of done**
- `organize` no longer requires the standing complexity exemption
- The refactor improves readability without increasing indirection everywhere else
- Existing behavior stays intact under test

**Why this issue is separate**
This issue is strictly maintainability work in the organizer core. It does not cover CI policy, coverage-ratchet tooling, or CLI smoke-test breadth.

### LINKED ISSUE #87: [FEATURE] Add CLI smoke coverage for high-traffic commands
**Problem**
The README promises broad CLI capability, but recent repo energy has gone into CI/test infrastructure. We still need a small, stable set of user-facing smoke paths around the highest-traffic commands.

**Goal**
Add reliable smoke coverage for the most important CLI workflows so regressions are caught closer to real user behavior.

**In scope**
- Choose a small set of top-priority commands, starting with `fo organize`, `fo search`, `fo dedupe`, and `fo copilot`
- Define the minimum viable success criteria for each smoke flow
- Add tests or fixtures that exercise these paths at a user-facing boundary
- Keep runtime reasonable enough for regular validation use

**Out of scope**
- Exhaustive end-to-end testing of every command and option
- Large fixture corpora or benchmark-style performance testing
- CI workflow trigger changes and coverage-baseline updates

**MECE plan**
1. Identify the 2-4 commands that represent the highest-value CLI trust surface.
2. Decide the narrowest smoke assertions that still catch real regressions.
3. Add shared fixtures/helpers so the smoke paths stay maintainable.
4. Wire the tests into the appropriate marker strategy without overloading the fast suite.
5. Document the intended smoke surface so future additions stay intentional.

**Definition of done**
- A small set of high-value CLI paths are exercised end-to-end or near-end-to-end
- The tests are stable enough for repeated CI usage
- The repo has a clear statement of which commands are part of the smoke contract

**Why this issue is separate**
This issue is about user-facing command confidence. It does not overlap with workflow triggers, local coverage tooling, or the organizer refactor.

### LINKED ISSUE #88: [FEATURE] Document triage policy for cleanup-only test work
**Problem**
Recent backlog noise has included low-signal test-cleanup work that is technically valid but not always worth standalone attention. Without a shared rule, small redundant cleanups can compete with higher-leverage CI and product hardening work.

**Goal**
Document a lightweight triage policy for cleanup-only test work so maintainers can consistently prioritize, defer, bundle, or close it.

**In scope**
- Define what qualifies as low-signal cleanup-only test work
- Document when to bundle that work into nearby changes versus taking it as a standalone issue/PR
- Add guidance for closing or deprioritizing redundant cleanup threads cleanly
- Place the guidance where contributors and maintainers will actually see it

**Out of scope**
- Rewriting the full contribution guide
- Changing test architecture directly
- Re-litigating every past cleanup PR one by one

**MECE plan**
1. Review a few recent examples of cleanup-only test work to identify the common pattern.
2. Write a short decision rule covering standalone-worthy, bundle-later, and close-as-redundant cases.
3. Add the guidance to the most appropriate maintainer-facing doc or workflow notes.
4. Apply the rule to the current queue as a sanity check.
5. Keep the guidance intentionally small so it is used, not ignored.

**Definition of done**
- Maintainers have a shared rule for triaging redundant test cleanups
- Low-value cleanup work stops competing with higher-impact priorities by default
- The policy is short, discoverable, and practical

**Why this issue is separate**
This issue is backlog-governance/process work. It does not change the CI workflows, local tooling, or application code itself.

### PR DESCRIPTION
**Pull request by curdriceaurora**
## Summary

Four independent improvements delivered as a single PR from \`codex/issues-85-88\`:

- **#88 — Triage policy doc**: New \`docs/developer/triage-policy.md\` with decision tree for cleanup-only test work. Added pointer in \`CONTRIBUTING.md\` and \`docs/developer/index.md\`.
- **#85 — One-command coverage ratchet**: New \`scripts/coverage/ratchet.sh\` wrapping the 3-step integration coverage flow into \`check\` / \`update\` / \`dry-run\` modes. Updated \`scripts/coverage/README.md\` and \`CONTRIBUTING.md\`.
- **#87 — CLI smoke tests**: New \`tests/cli/test_cli_smoke.py\` with 4 smoke tests (\`fo organize\`, \`fo search\`, \`fo dedupe scan\`, \`fo copilot status\`). All marked \`smoke + ci + unit\`, run in 1.25s.
- **#86 — Organizer refactor (C901 removal)**: Extracted 5 private methods from \`organize()\` (complexity 33 → ≤15): \`_categorize_files\`, \`_process_image_type\`, \`_process_all_file_types\`, \`_deduplicate_processed\`, \`_execute_organization\`. Removed C901 exemption from \`pyproject.toml\`. Added \`TestCategorizeFiles\` (4 tests), \`TestDeduplicateProcessed\` (3 tests), \`TestProcessImageType\` (3 tests).

## Closes

Closes #85
Closes #86
Closes #87
Closes #88

## Test plan

- [x] \`pymarkdown scan docs/developer/triage-policy.md\` → 0 violations
- [x] \`bash scripts/coverage/ratchet.sh unknown_mode\` → exits 2 with usage message
- [x] \`bash scripts/coverage/ratchet.sh check\` → validates baseline; \`update\` ratchets upward; \`dry-run\` previews changes
- [x] \`pytest -m smoke tests/cli/test_cli_smoke.py -q --override-ini="addopts="\` → 4 passed
- [x] \`ruff check --select C901 src/file_organizer/core/organizer.py\` → no violations
- [x] \`pytest tests/core/test_organizer.py -q --override-ini="addopts="\` → 33 passed
- [x] Pre-commit hooks: all pass
- [x] CI: Test PR suite (py3.11) → SUCCESS

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -163,6 +163,9 @@ ruff check src/
 
 # Type check
 mypy src/file_organizer/ --strict
+
+# Ratchet integration coverage baseline (after improving integration tests)
+bash scripts/coverage/ratchet.sh update
```
[the remark this answers is not in the document]
Comment (curdriceaurora) [reply CONTRIBUTING.md]: ✅ Fixed: added `bash scripts/coverage/ratchet.sh check` (read-only validation) as the primary example before `update`, with a comment clarifying that `update` only mutates the baseline.

[the remark this answers is not in the document]
Comment (curdriceaurora) [reply CONTRIBUTING.md]: Fixed: added check mode as the primary example before update in CONTRIBUTING.md.

File: scripts/coverage/ratchet.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+MODE="${1:-check}"
+
+case "${MODE}" in
+  check|update|dry-run) ;;
+  *)
+    echo "ERROR: unknown mode '${MODE}'" >&2
+    echo "Usage: bash scripts/coverage/ratchet.sh [check|update|dry-run]" >&2
+    exit 2
+    ;;
+esac
```
[the remark this answers is not in the document]
Comment (curdriceaurora) [reply scripts/coverage/ratchet.sh:32]: Fixed: added extra-args guard before mode validation in ratchet.sh.

File: tests/cli/test_cli_smoke.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+class TestOrganizeSmoke:
+    """fo organize --dry-run exits 0 and reports processed count."""
+
+    # FileOrganizer is lazy-imported inside organize(); patching at definition
+    # site because cli.organize holds no module-level reference to the class.
+    @patch("file_organizer.core.organizer.FileOrganizer")
+    @patch(_SETUP_PATCH, return_value=True)
+    def test_organize_dry_run_exits_zero(
+        self, _setup: MagicMock, mock_cls: MagicMock, tmp_path: Path
```
[the remark this answers is not in the document]
Comment (curdriceaurora) [reply tests/cli/test_cli_smoke.py]: Fixed: use new=MagicMock on @patch to avoid injecting unused param, eliminating PT019.

File: tests/cli/test_cli_smoke.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+class TestSearchSmoke:
+    """fo search glob exits 0 and names at least one match."""
+
+    def test_search_glob_exits_zero(self, tmp_path: object) -> None:
+        import pathlib
+
+        d = pathlib.Path(str(tmp_path))
+        (d / "report.txt").write_text("hello")
+        (d / "notes.txt").write_text("world")
```
[the remark this answers is not in the document]
Comment (curdriceaurora) [reply tests/cli/test_cli_smoke.py]: Fixed: tmp_path: Path in TestSearchSmoke, dropped pathlib conversions.

File: tests/cli/test_cli_smoke.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        mock_det = MagicMock()
+        mock_get_detector.return_value = mock_det
+        mock_det.get_duplicate_groups.return_value = {}
+
+        result = runner.invoke(app, ["dedupe", "scan", str(tmp_path)])
+
+        assert result.exit_code == 0, result.output
+        assert "No duplicates" in result.output
+        mock_det.scan_directory.assert_called_once_with(Path(str(tmp_path)), ANY)
+
```
[the remark this answers is not in the document]
Comment (curdriceaurora) [reply tests/cli/test_cli_smoke.py]: Fixed: tmp_path: Path in TestDedupeScanSmoke, simplified assertion.

File: src/file_organizer/core/organizer.py [hunk 6]
```diff
@@ -189,7 +190,58 @@ def organize(
...
+            other_files=other_files,
+        )
+
+        all_processed = self._process_all_file_types(
+            text_files, image_files, video_files, audio_files, cad_files
+        )
+
+        if all_processed:
+            all_processed = self._deduplicate_processed(all_processed, result)
+            failed_cnt = len([p for p in all_processed if p.error])
```
[the remark this answers is not in the document]
Comment (curdriceaurora) [reply src/file_organizer/core/organizer.py]: Fixed: sum(1 for p in all_processed if p.error) replaces list allocation.

File: src/file_organizer/core/organizer.py [hunk 7]
```diff
@@ -301,88 +353,82 @@ def organize(
...
+                with pf.file_path.open("rb") as f:
+                    for chunk in iter(lambda: f.read(65536), b""):
+                        hasher.update(chunk)
+                file_hash = hasher.hexdigest()
+            except OSError:
+                deduped.append(pf)
+                continue
+            if file_hash not in seen_hashes:
+                seen_hashes[file_hash] = pf
+                deduped.append(pf)
```
[the remark this answers is not in the document]
Comment (curdriceaurora) [reply src/file_organizer/core/organizer.py]: Fixed: seen_hashes is now set[str], uses .add(), no stored values.
